src/watch/index.ts

```diff
@@ -208,29 +234,39 @@ export const watchCommand = command({
 	 * As an alternative, we watch cwd and all run-time dependencies
 	 */
 	const watcher = watch(
-		[
-			...argv._,
-			...options.include,
-		],
+		argv._,
 		{
 			cwd: process.cwd(),
 			ignoreInitial: true,
-			ignored: [
-				// Hidden directories like .git
-				'**/.*/**',
-
-				// Hidden files (e.g. logs or temp files)
-				'**/.*',
-
-				// 3rd party packages
-				'**/{node_modules,bower_components,vendor}/**',
-
-				...options.exclude,
-			],
 			ignorePermissionErrors: true,
+			// ignore all files that are by default ignored or explicitly excluded
+			ignored: file => isDefaultIgnore(file) || isOptionsExclude(file),
 		},
 	).on('all', reRun);
 
+	if (resolvedIncludes.length > 0) {
+		const globParents = resolvedIncludes.map(pattern => (
+			isGlob(pattern)
```

Comment:
This seems like a pretty [hefty function](https://github.com/micromatch/is-glob/blob/master/index.js) to run only to check if it's a glob. What if we just run `globParent` on all inputs regardless of whether they're a glob?

Reply:
I tried that but a simple include like test/foo/hello becomes test/foo when run through glob-parent. We could proactively append a separator at the end but that may result in weirder behavior potentially. Coincidentally glob-parent uses isGlob under the hood.